runtime/src/bank/check_transactions.rs

```diff
@@ -89,14 +91,21 @@ impl Bank {
             .iter()
             .zip(lock_results)
             .map(|(tx, lock_res)| match lock_res {
-                Ok(()) => self.check_transaction_age(
-                    tx.borrow(),
-                    max_age,
-                    &next_durable_nonce,
-                    &hash_queue,
-                    next_lamports_per_signature,
-                    error_counters,
-                ),
+                Ok(()) => {
+                    let compute_budget_limits = process_compute_budget_instructions(
+                        tx.borrow().program_instructions_iter(),
+                        &self.feature_set,
+                    );
```

Comment:
I don't think it belongs in `fn check_age` either..

Reply:
I had to rename `check_age` for this, since it's iterating over all the transactions in the slice in this function. Doing it outside would mean another iteration over the transactions. Hope this works.